akka-http-core/src/main/scala/akka/http/model/MediaType.scala

```diff
@@ -253,33 +274,34 @@ object MediaTypes extends ObjectRegistry[(String, String), MediaType] {
 
   def forExtension(ext: String): Option[MediaType] = extensionMap.get(ext.toRootLowerCase)
 
-  private def app(subType: String, compressible: Boolean, binary: Boolean, fileExtensions: String*) = register {
-    new NonMultipartMediaType("application", subType, compressible, binary, immutable.Seq(fileExtensions: _*)) {
+  private def app(subType: String, compressible: Boolean, encoding: Encoding, fileExtensions: String*) = register {
+    new NonMultipartMediaType("application", subType, compressible, encoding, immutable.Seq(fileExtensions: _*)) {
       override def isApplication = true
     }
   }
   private def aud(subType: String, compressible: Boolean, fileExtensions: String*) = register {
-    new NonMultipartMediaType("audio", subType, compressible, binary = true, immutable.Seq(fileExtensions: _*)) {
+    new NonMultipartMediaType("audio", subType, compressible, encoding = Encoding.Binary, immutable.Seq(fileExtensions: _*)) {
       override def isAudio = true
     }
   }
-  private def img(subType: String, compressible: Boolean, binary: Boolean, fileExtensions: String*) = register {
-    new NonMultipartMediaType("image", subType, compressible, binary, immutable.Seq(fileExtensions: _*)) {
+  private def img(subType: String, compressible: Boolean, encoding: Encoding, fileExtensions: String*) = register {
+    new NonMultipartMediaType("image", subType, compressible, encoding, immutable.Seq(fileExtensions: _*)) {
       override def isImage = true
     }
   }
   private def msg(subType: String, fileExtensions: String*) = register {
-    new NonMultipartMediaType("message", subType, compressible = true, binary = false, immutable.Seq(fileExtensions: _*)) {
+    new NonMultipartMediaType("message", subType, compressible = true, encoding = Encoding.Fixed(HttpCharsets.`UTF-8`),
```

Comment:
AFAICS the old RFC 2046 is the relevant here and defines content to be generally US-ASCII encoded.
Since UTF8 is a super-set of US-ASCII I went for UTF8 here, since it appears to be the best default choice nowadays.